### ISSUE TITLE
Make old "go to the current version of this document" from lore docs point at new sphinx docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6926]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#6926|
|Type|release blocker: regression|
|Created|2014-01-21 19:36:38Z|
|Branch|https://github.com/twisted/twisted/tree/fix-doc-locations-6926-2|

As noted [ticket:4500#comment:46 on the original ticket], the current directory structure has been muddled around a bit, mostly by the addition of the somewhat spurious "projects" directory.  We should rearrange the directory structure (and update the toctrees to reflect the re-organization) so that the URLs line up.  As a bonus, the 'projects' tree is just unnecessarily deep nesting which probably doesn't provide any value anyway.

<details><summary>Searchable metadata</summary>

```
trac-id__6926 6926
type__release_blocker__regression release blocker: regression
reporter__glyph glyph
priority__high high
milestone__Twisted_14_0_0 Twisted-14.0.0
branch__branches_fix_doc_locations_6926_2 branches/fix-doc-locations-6926-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1390332998000000 1390332998000000
changetime__1396981384000000 1396981384000000
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

We need to take care of this before release.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set owner to @hawkowl|
|-|-|

I'll work on it tonight.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

(In [41966]) Branching to fix-doc-locations-6926.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

Okay, gave it a shot.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

I forced a build of the branch.  The documentation builder is green but unfortunately it still doesn't actually produce a working build of the documentation.

Apart from that, I wonder how to determine if this change addresses the issue or not?  I suppose I should manually verify that the files generated by sphinx now have the same paths (and therefore will exist at the same URLs) as the files generated by lore before the conversion.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

Another issue here is how we will generate these links in the Sphinx documentation.  Has that been implemented?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

I think the links aren't appearing because the doc builder has Sphinx 1.1, which doesn't support hidden toc trees or something.

Let's see if it works when I set 1.2 required, forced build...

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

Yep, it fails. To build working docs, the buildslave needs Sphinx 1.2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl commented|
|-|-|

Done a PR for it here: https://github.com/twisted-infra/buildslave-config/pull/5

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl commented|
|-|-|

Tom Prince merged and update the buildslaves, it now goes all green.

The buildslave also creates a working build for you now, exarkun :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set owner to @hawkowl|
|-|-|

> Another issue here is how we will generate these links in the Sphinx documentation. Has that been implemented?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_15"></a>|@hawkowl removed owner|
|-|-|

This actually does it now! I used the same code as from lore's templates, put the link where I thought it'd be good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

Why do both `doc` and `docs` still exist in this branch?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_18"></a>|@hawkowl removed owner|
|-|-|

Nuked the doc/ directory - (for future reference: git doesn't track empty dirs, and didn't delete them when [#4500](https://github.com/twisted/twisted/issues/4500) deleted all the files).

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_19"></a>|Alex set owner to @hawkowl|
|-|-|

This looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_20"></a>|@hawkowl commented|
|-|-|

(In [42139]) Branching to fix-doc-locations-6926-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_21"></a>|@hawkowl set status to `closed`|
|-|-|

(In [42144]) Merge fix-doc-locations-6926-2

Author: hawkowl
Reviewers: exarkun, Alex
Fixes: [#6926](https://github.com/twisted/twisted/issues/6926)

Fix the doc locations and add "go to current version of this document" to Sphinx built docs.
